hss/src/hssStatsPromClient.cpp

```diff
@@ -0,0 +1,621 @@
+mme_msg_tx_s6a_auth_info_answer_success(mme_msg_tx_family.Add({{"interface","s6a"},{"msg_type","auth_info_answer"},{"result","success"}})),
+mme_msg_tx_s6a_auth_info_answer_failure(mme_msg_tx_family.Add({{"interface","s6a"},{"msg_type","auth_info_answer"},{"result","failure"}})),
+mme_msg_tx_s6a_update_location_answer_success(mme_msg_tx_family.Add({{"interface","s6a"},{"msg_type","update_location_answer"},{"result","success"}})),
+mme_msg_tx_s6a_update_location_answer_failure(mme_msg_tx_family.Add({{"interface","s6a"},{"msg_type","update_location_answer"},{"result","failure"}})),
+mme_msg_tx_s6a_purge_answer_success(mme_msg_tx_family.Add({{"interface","s6a"},{"msg_type","purge_answer"},{"result","success"}})),
+mme_msg_tx_s6a_purge_answer_failure(mme_msg_tx_family.Add({{"interface","s6a"},{"msg_type","purge_answer"},{"result","failure"}})),
+mme_msg_tx_s6a_cancel_location_request(mme_msg_tx_family.Add({{"interface","s6a"},{"msg_type","cancel_location_request"}}))
+{
+}
+
+
+mme_msg_tx_counters::~mme_msg_tx_counters()
+{
+}
+
+
+
+
+mme_msg_rx_counters::mme_msg_rx_counters():
+mme_msg_rx_family(BuildCounter().Name("number_of_messages_received").Help("Number of messages received by HSS ").Labels({{"direction","incoming"}}).Register(*registry)),
+mme_msg_rx_s6a_auth_info_request(mme_msg_rx_family.Add({{"interface","s6a"},{"msg_type","auth_info_request"}})),
+mme_msg_rx_s6a_update_location_request(mme_msg_rx_family.Add({{"interface","s6a"},{"msg_type","update_location_request"}})),
+mme_msg_rx_s6a_purge_request(mme_msg_rx_family.Add({{"interface","s6a"},{"msg_type","purge_request"}}))
+{
+}
+
+
+mme_msg_rx_counters::~mme_msg_rx_counters()
+{
+}
+
+
+
+
+void hssStats::increment(hssStatsCounter name,std::map<std::string,std::string> labels)
+{
+	switch(name) {
+	case hssStatsCounter::MME_MSG_TX_S6A_AUTH_INFO_ANSWER_SUCCESS:
+	{
+		mme_msg_tx_m->mme_msg_tx_s6a_auth_info_answer_success.Increment();
+		if(labels.size() == 0) {
+		break;
+		}
+		if(labels.size() == 1) {
+		auto it = labels. begin();
+		struct Node s1 = {name, it->first, it->second};
+		auto it1 = metrics_map.find(s1);
+		if(it1 != metrics_map.end()) {
+		    mme_msg_tx_DynamicMetricObject1 *obj = static_cast<mme_msg_tx_DynamicMetricObject1 *>(it1->second);
+		    obj->counter.Increment();
+		} else {
+		    mme_msg_tx_DynamicMetricObject1 *obj = mme_msg_tx_m->add_dynamic1("interface","s6a","msg_type","auth_info_answer","result","success",it->first, it->second);
+		    auto p1 = std::make_pair(s1, obj);
+		    metrics_map.insert(p1);
+		    obj->counter.Increment();
+		}
+		} else if (labels.size() == 2) {
+		auto it1 = labels. begin();
+		auto it2 = it1++;
+		struct Node s1 = {name, it1->first+it2->first, it2->second+it2->second};
+		auto itf = metrics_map.find(s1);
+		if(itf != metrics_map.end()) {
+		    mme_msg_tx_DynamicMetricObject2 *obj = static_cast<mme_msg_tx_DynamicMetricObject2 *>(itf->second);
```

Comment:
can we add function for this label handling to avoid repeated code.?

Reply:
Yeah..this code is auto generated for the same reason. Wanted to avoid repeated coding the same thing. We just write json file and hssStatsPromClient.cpp and hssStatsPromClient.h are autogenerated.